apis/python/src/tiledbsoma/pytiledbsoma.cc

```diff
@@ -50,659 +50,727 @@ using namespace py::literals;
+                           dim, coords.cast<std::vector<int32_t>>());
+                   } else if (!strcmp(arrow_schema.format, "s")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<int16_t>>());
+                   } else if (!strcmp(arrow_schema.format, "c")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<int8_t>>());
+                   } else if (!strcmp(arrow_schema.format, "L")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<uint64_t>>());
+                   } else if (!strcmp(arrow_schema.format, "I")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<uint32_t>>());
+                   } else if (!strcmp(arrow_schema.format, "S")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<uint16_t>>());
+                   } else if (!strcmp(arrow_schema.format, "C")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<uint8_t>>());
+                   } else if (!strcmp(arrow_schema.format, "f")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<float>>());
+                   } else if (!strcmp(arrow_schema.format, "g")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<double>>());
+                   } else if (
+                       !strcmp(arrow_schema.format, "u") ||
+                       !strcmp(arrow_schema.format, "z")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<std::string>>());
+                   } else if (
+                       !strcmp(arrow_schema.format, "tss:") ||
+                       !strcmp(arrow_schema.format, "tsm:") ||
+                       !strcmp(arrow_schema.format, "tsu:") ||
+                       !strcmp(arrow_schema.format, "tsn:")) {
+                       // convert the Arrow Array to int64
+                       auto pa = py::module::import("pyarrow");
+                       coords = array.attr("cast")(pa.attr("int64")()).attr("tolist")();
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<int64_t>>());
+                   } else if (
+                       !strcmp(arrow_schema.format, "U") ||
+                       !strcmp(arrow_schema.format, "Z")) {
+                       reader.set_dim_points(
+                           dim, coords.cast<std::vector<std::string>>());
+                   } else {
+                       throw TileDBSOMAError(fmt::format(
+                           "[pytiledbsoma] set_dim_points: type={} not "
+                           "supported",
+                           arrow_schema.format));
```

Comment:
We need to remove `fmt` formatting from the Pybind11 code. Just doing basic string concatenation here is fine.